src/test/scala/scalang/NodeSpec.scala

```diff
@@ -104,6 +104,8 @@ class NodeSpec extends SpecificationWithJUnit {
         node.send(('echo, echo_host), mbox.self, (mbox.self, msg))
         mbox.receive(receiveTimeout) == (Some(msg))
       }) must beTrue
+      node.send(('echo, echo_host), mbox.self, (mbox.self, 'stop))
```

Comment:
It would be better to do the cleanup in the `doAfter`. This would ensure the echo subprocess is killed on test failure. I see two approaches how it can be implemented:

1. Create `class NodeSpecEcho extends SpecificationWithJUnit {}` with it's own `doAfter` block
2. Use global variable
   - add `var defered: List[() => Unit] = List()`
   - append `cleanup` fixtures to it 
     ```
     val cleanup = {
        node.send(....)
        erl.waitFor
     }
     defered =  cleanup :: defered
     ```
   - finally iterate over `defered` in `doAfter` and call each cleanup callback

I think the first approach is easier though.

Reply:
The `destroy` method is not always reliable. It uses SIGTERM signal to kill, which might be ignored by the subprocess. In fact erlang was ignoring it till Erlang 19.3 (I am not sure how it works now). Also `destroy` doesn't work if you spawn another process. I am pretty sure `#!/usr/bin/env escript` starts a new process. Which means `destroy` is effectively `nop`. That is why I proposed to handle `stop` event.

Reply:
I understand that `destroy` is not reliable, and I think using a `stop` message to terminate the escript OS process is a good idea, but sending a `stop` in a `doAfter` would not be straightforward since it uses mailboxes that are set up in the body of the test.